### ISSUE TITLE
Add new data type - delegate(🏃🏻)

### ISSUE THREAD

**Issue by karolstawowski** [TodePond/DreamBerd#47]
A delegate(🏃🏻) is a type that represents references to ~~methods~~ anything. When you instantiate a delegate, you can associate its instance with any ~~method~~ thing ~~with a compatible signature and return type~~. You can  call ~~the method~~ everything through the delegate instance. Unlike Microsoft Java delegates, DreamBerd delegates return a random thing out of all things passed as a delegate's instance parameter.

Example of usage:
```java
const const 👷🏻‍♂️: 🏃🏻 = new 🏃🏻()!!!

const const 🏎: Char[] = "Max"!
const const 🐢: Char[] = "Lewis"!
const const one: Int3 = 1!
const const truth: Boolean = True!
const const dateNow: Number = Date.now()!
fn add (a, b) => a + b!

var var 🕴🏻 = 👷🏻‍♂️(🏎)!
🕴🏻 = 👷🏻‍♂️(🐢)!
🕴🏻 = 👷🏻‍♂️(one)!
🕴🏻 = 👷🏻‍♂️(truth)!
🕴🏻 = 👷🏻‍♂️(dateNow)!
🕴🏻 = 👷🏻‍♂️(add)!

print(🕴🏻)! // prints random value out of things passed - "Max", "Lewis", 1, True, 1672872867639 or Error: Can't cast a funct without providing arguments!
```

Delegate add please. Reference good. Value bad.

**Comment by TodePond** [comment]
Hey I get that reference! 

Microsoft Java is actually called 'D flat' right?

**Comment by karolstawowski** [comment]
```java
const const 👷🏻‍♂️: 🏃🏻 = new 🏃🏻()!!!

var var 🕴🏻 = 👷🏻‍♂️("Microsoft Java" ==== "D flat")!

print(🕴🏻)!!! // prints True
```

**Comment by karolstawowski** [comment]
@TodePond what do you think about adding the **delegates**(🏃🏻🏃🏻)?

**Comment by TodePond** [comment]
Yes, but I'm still thinking about how we could milk the joke even more 🤔

We could have a section about emojis being used as variable names.

And then a section about delegates

**Comment by karolstawowski** [comment]
Yeah, adding emojis as a variable names would be a nice feature. I've made a simple template, which could be expanded with better ideas.

## Variables
Variables can be named with any Unicode character or string.
```java
const const firstAlphabetLetter = 'A'!
const var 🕳 = "Programmer's brain"!
var const 🇺🇸 = "Shooting range"!
const const 👍 = True!
var var 1️⃣ = 1! 
```

Some of Unicode characters are keywords which can not be used as variable names.
```java
🏃🏻 <- Delegate
📅 <- Date
❗ <- Exclamation mark
```
Disclaimer: *Well, actually you can use all Unicode characters as variable names, but don't tell anyone.*

Then we can place delegates section somewhere below. 

Is it okay?

**Comment by TodePond** [comment]
This is looking good! I'd get rid of the american line because that seems to be a little bit distasteful to me. Would you like to do a PR? I'll then do a final edit on it.

**Comment by karolstawowski** [comment]
Here's a pull request regarding variables and constants: #48

**Comment by karolstawowski** [comment]
Project manager is asking in which sprint this will be added. Our codebase is DreamBerd heavy and our devs are really excited about the emojis feature (at least that's what PM is saying, along something with morale boost and happier environment after layoffs). We will be announcing this with our TikTok integration.

**Comment by TodePond** [comment]
Sorry there were some layoffs (ie: tode fell asleep) and I was pulled onto this: https://github.com/TodePond/MeanPond/issues/1

I'll merge this tomorrow!

**Comment by karolstawowski** [comment]
Okay! After merging variables and constants documentation we have got to move on to creation of delegates. 👨‍💻

**Comment by TodePond** [comment]
I tweaked it a bit and merged! Next up is 'special emojis' like the delegate emoji. I'm thinking that we should be able to use any 'person' emoji for this.

eg: 🏃, 🎅, 👯‍♂️

**Comment by karolstawowski** [comment]
By saying that, do you mean something like this?

## Special emojis
Some of Unicode characters are keywords which can not be used as variable names. 
```java
📅 <- Date
🎲 <- Random
✔️ <- True
🤔 <- Maybe
🟰 <- Equals mark
❗ <- Exclamation mark
```

People emojis are reserved for delegates, e.g.: 🏃, 🎅, 👯‍♂️, 🕵️

Disclaimer: *Well, actually you can use all Unicode characters as variable names, but don't tell anyone.*

## Delegates
Delegate is a type that represents references to anything. When you instantiate a delegate, you can associate its instance with anything. You can call everything through the delegate instance. Delegates return a random thing out of all the things passed as a delegate's instance parameter.
```java
const const 👷🏻‍♂️: 🏃🏻 = new 🏃🏻()!!!

const const 🏎: Char[] = "Max"!
const const 🐢: String = "Lewis"!
const const one: Int3 = 1️⃣!
const const truth: Boolean = ✔️!
const const dateNow: Number = 📅.now()!
fn add (a, b) => a + b!

var var 🕴🏻 = 👷🏻‍♂️(🏎)!
🕴🏻 = 👷🏻‍♂️(🐢)!
🕴🏻 = 👷🏻‍♂️(one)!
🕴🏻 = 👷🏻‍♂️(truth)!
🕴🏻 = 👷🏻‍♂️(dateNow)!
🕴🏻 = 👷🏻‍♂️(add)!

print(🕴🏻)!!! // prints random value out of things passed - "Max", "Lewis", 1, True, 1675353693 or Error: Can't cast a fun without providing arguments!
```
